retdec: 3.2 -> 5.0

katrinafyi opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
https://github.com/avast/retdec/blob/master/CHANGELOG.md

- Bumps vendored dependencies and remove ones no longer needed.
- Since 3.3, compiled patterns are not shipped in the support file, obviating the postFetch strip. (https://github.com/avast/retdec-support/issues/3)
- Now, patterns may be compiled at build time and an argument is provided to control this (on by default).
  - As such, retdec-full is no longer needed and removed. The 60MB increase from the compiled patterns seems more preferred than duplicating the whole 500MB size.
- We use cmake _URL variables to insert dependencies and we are able to use nixpkgs googletest.
- Fix build with current gcc 13.
- Remove i686 from platforms as derivation script needs to specify lib64.


On my laptop, this package takes 30 minutes to build. Half of this is the fork of LLVM.
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->
cc maintainers: @timokau @dtzWill 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

timokau commented on pkgs/development/tools/analysis/retdec/default.nix:
```diff
@@ -15,131 +15,131 @@
+  patches = [
+    # gcc 13 compatibility
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/avast/retdec/pull/1153.patch";
```
Use a stable link that refers to a commit hash instaed:

https://github.com/avast/retdec/commit/dbaab2c3d17b1eae22c581e8ab6bfefadf4ef6ae.patch

timokau reviewed (changes requested):
Thanks a lot for this!

I am not using this package anymore and therefore cannot maintain it. Would you be willing to become a maintainer? If so, please add yourself to the maintainers list and remove my entry.

katrinafyi committed:
maintainers: add katrinafyi

katrinafyi committed:
retdec: 3.2 -> 5.0

- Bumps vendored dependencies and remove ones no longer needed.
- Since 3.3, compiled patterns are not shipped in the support file, obviating the postFetch strip. (avast/retdec-support#3)
- Now, patterns may be compiled at build time and an argument is provided to control this (on by default).
- As such, retdec-full is no longer needed and removed. The 60MB increase seems more preferred than duplicating the 500MB size.
- We use cmake _URL variables to insert dependencies and we are able to use nixpkgs googletest.
- Fix build with current gcc 13.
- Remove i686 from platforms, as derivation needs to specify lib64.
- Maintainers: remove timokau, add katrinafyi.

katrinafyi commented:
Addressed changes (use absolute patch url and change maintainers) @timokau 

I had a question, does anything else need to be done to obsolete the retdec-full package?

katrinafyi commented:
Fixed build failure due to stale hash.

timokau commented on maintainers/maintainer-list.nix:
```diff
@@ -9661,6 +9661,11 @@
+    name = "katrinafyi";
+    github = "katrinafyi";
+    githubId = 39479354;
+  };
```
Welcome to the maintainers team :tada:

katrinafyi committed:
retdec-full: remove

since version 5.0, compiled yara patterns are included with the retdec package
so retdec-full is no longer needed.

katrinafyi commented:
Formally remove retdec-full.

timokau reviewed (approved):
Thank you! Quite an impressive first contribution :)

kirillrdy commented on pkgs/development/tools/analysis/retdec/default.nix:
```diff
@@ -159,64 +160,66 @@ in stdenv.mkDerivation rec {
+        --replace-quiet VersionTests DISABLED_VersionTests
+
+      substituteInPlace scripts/retdec-utils.py \
+        --replace-warn /usr/bin/time ${time} \
```
any reason not to use `--replace-fail` here ?

katrinafyi replied:
Not really, and it would be good to fail more obviously here. I have an open PR for this package (#293962) and I can make the changes there.
